File: Baekjoon/16435.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);
  cout.tie(0);
  
  int n, fruit, snakeLen;
  vector<int> fruits;

  cin >> n >> snakeLen;
  for (int i = 0; i < n; i++) {
    cin >> fruit;
    fruits.push_back(fruit);
  }
  
  sort(fruits.begin(), fruits.end());
  vector<int>::iterator iter;

  for (iter = fruits.begin(); iter != fruits.end(); iter++) {
    if (*iter <= snakeLen) snakeLen++;
    else break;
  }

  cout << snakeLen << '\n';

  return 0;
}

File: python/solutions/16435.py
import sys

n, snake_len = map(int, sys.stdin.readline().split(' '))
fruits = list(map(int, sys.stdin.readline().split(' ')))

fruits.sort()

for fruit in fruits:
    if snake_len >= fruit:
        snake_len += 1
    else:
        break
print(snake_len)